lib/rspec-puppet/cache.rb

```diff
@@ -11,10 +11,10 @@ def initialize(&default_proc)
     end
 
     def get(*args, &blk)
+      expire!
       if !@cache.has_key? args
         @cache[args] = (blk || @default_proc).call(*args)
         @lra << args
-        expire!
```

Comment:
if `expire!` is kept here, this cache can be a LRU cache by doing this on the line above:
```
@irl = (@irl << 1).reverse.uniq.reverse
```
As an example given [1,2,3,4] and appending 2, the result is [1,3,4,2]. A long as cache is a small number of entries that solution is good enough. Then, when expires removes entries, it will remove the MAX_ENTRIES least recently used entries